index.js

```diff
@@ -1,47 +1,136 @@
-module.exports = MediaSourceStream
+module.exports = MediaElementWrapper
 
+var EventEmitter = require('events')
 var inherits = require('inherits')
 var stream = require('stream')
+var toArrayBuffer = require('to-arraybuffer')
 
 var MediaSource = typeof window !== 'undefined' && window.MediaSource
 
-inherits(MediaSourceStream, stream.Writable)
+var DEFAULT_BUFFER_DURATION = 60 // seconds
 
-function MediaSourceStream (elem, opts) {
+function MediaElementWrapper (elem, opts) {
   var self = this
-  if (!(self instanceof MediaSourceStream)) return new MediaSourceStream(elem, opts)
-  stream.Writable.call(self, opts)
+  if (!(self instanceof MediaElementWrapper)) return new MediaElementWrapper(elem, opts)
+  EventEmitter.call(self)
 
   if (!MediaSource) throw new Error('web browser lacks MediaSource support')
-  if (!opts) opts = {}
 
+  self._opts = opts || {}
   self._elem = elem
   self._mediaSource = new MediaSource()
-  self._sourceBuffer = null
-  self._cb = null
-
-  self._type = opts.type || getType(opts.extname)
-  if (!self._type) throw new Error('missing `opts.type` or `opts.extname` options')
+  self._streams = []
 
   self._elem.src = window.URL.createObjectURL(self._mediaSource)
+}
+
+inherits(MediaElementWrapper, EventEmitter)
```

Comment:
actually it doesn't need to inherit from `EventEmitter` anyway. I'll just get rid of this